kvs/client/python/anna/ipc_client.py

```diff
@@ -132,30 +130,25 @@ def causal_get(self, keys, future_read_set,
             request.versioned_key_locations[addr].versioned_keys.extend(
                                 versioned_key_locations[addr].versioned_keys)
 
-        for keys in keys:
+        for key in keys:
             tp = request.tuples.add()
             tp.key = key
 
         request.response_address = self.get_response_address
 
-        (request.future_read_set.add(k) for k in future_read_set)
+        request.future_read_set.extend(future_read_set)
 
         self.get_request_socket.send(request.SerializeToString())
 
+
         try:
             msg = self.get_response_socket.recv()
         except zmq.ZMQError as e:
             if e.errno == zmq.EAGAIN:
                 logging.error("Request for %s timed out!" % (str(keys)))
             else:
                 logging.error("Unexpected ZMQ error: %s." % (str(e)))
-
-            resp = {}
```

Comment:
This change seems incorrect. If we're expecting a map, we should not return `None` but instead return a map of keys mapped to `None`s.

Reply:
in `call.py`, the executor first check if it is `None` and if not, it expects a map structure and proceed.